### PR TITLE
Update 

### PR DESCRIPTION
**Pull request by rnikoopour**
Changes support syntax highlighting for newer versions of Terraform

### CODE REVIEW

File: terraform-mode.el [hunk 1]
```diff
@@ -43,8 +43,21 @@
...
+      (concat "^\\s-*\\(" (string-join terraform--builtin-blocks "\\|") "\\)\\s-+"))
+
+(defconst terraform--builtin-inline-regexp
+  (concat "\\(" (string-join terraform--builtin-inline "\\|") "\\)\\."))
+
+(defconst terraform--inner-inline-regexp
+  "\\.\\(.*?\\)\\.")
+
+(defconst terraform--resource-regexp
+      (concat "^\\s-*\\(" (string-join terraform--builtin-blocks "\\|") "\\)\\s-+\\(\"?.*?\"?\\)\\s-+"))
```
Comment (tarsius) [inline terraform-mode.el:60]: Be more careful not to mess up whitespace. (One more above.)

**Comment by tarsius** [comment]
I guess you replaced the `defconst` with `setq` to ease trying out the new values.

1. You don't need to do that. Use `eval-defun` to reset a variable definition. For `defconst` that isn't even necessary, here `eval-last-sexp` works too.

2. Do not commit the fact that you did that and then undo it using another commit. That's not history that is work preserving. Even if create a first commit that replaces `defconst` with `setq` for some reason, then still don't use a second commit to undo it. Instead undo the change and the amend that to the original commit.

**Comment by rnikoopour** [comment]
Thank you for the feedback I will roll back this last commit and amend it.

**Comment by rnikoopour** [comment]
@tarsius I'm going to close this PR and open a draft PR with a new set of commits.  It'll have a cleaner history and I'm leveraging `rx`.  Since this commit history was already polluted, I don't think it's worth trying to clean up.

**Pull request closed without merging**